content/doc/book/pipeline/syntax.adoc

```diff
@@ -293,7 +293,10 @@ was successful.
 run has an "aborted" status, usually due to the Pipeline being manually aborted.
 This is typically denoted by gray in the web UI.
 `failure`:: Only run the steps in `post` if the current Pipeline's or stage's
-run has a "failed" status, typically denoted by red in the web UI.
+run has a "failed" status, typically denoted by red in the web UI. Note that if
+you manually set `currentBuild.result = 'FAILURE'` in a stage and have a
+`failure` `post` condition on that stage, the `failure` will not fire for that
+stage.
```

Comment:
Should we tell users what to do instead of setting `currentBuild.result = 'FAILURE'`? Would switching to using the `error` step do the right thing?

Reply:
Good question - I feel like going into detail on that here is probably not the right place, though.

Reply:
If there's a blog post coming along with 1.3.2, maybe there? 

(Disclaimer: I don't know that there's a blog post coming or not :) )

Reply:
> I feel like going into detail on that here is probably not the right place, though.

Fair enough. My other thought is that it seems unlikely that anyone who has an issue would think to look in the docs to see if anything changed, and AIUI we don't really want users to be setting `currentBuild.result` manually in script blocks in the first place, so maybe it would be better to not change the docs at all?

I guess ideally we'd have general documentation about the best way to set the build result and raise errors in declarative, which would encourage new users to do the right thing, but creating that is definitely more work 😄

I would be fine with just mentioning the change in the release notes, since that seems like the most likely place that anyone having issues would check, but I am fine with documenting it as well if you think it will be helpful to new users as well.

Reply:
Hmmmmmmm. Well, we can leave the docs as is for the moment and continue thinking about the right venue, wording, etc - there's no reason we need to update the docs immediately after the release. Might be worth holding in this in reserve in case there's confusion once it's released?